### PR TITLE
removed extra semicolon after function definition

### PR DESCRIPTION
**Pull request by tarptaeya**

### CODE REVIEW

File: templates/cpp-template-default/Classes/HelloWorldScene.h [hunk 1]
```diff
@@ -38,7 +38,7 @@ class HelloWorld : public cocos2d::Scene
     void menuCloseCallback(cocos2d::Ref* pSender);
     
     // implement the "static create()" method manually
-    CREATE_FUNC(HelloWorld);
```
Comment (jcs090218) [inline templates/cpp-template-default/Classes/HelloWorldScene.h:41]: I thought the semicolon is for the better readability?

**Review by jcs090218 — COMMENTED** [review]
Double checking the PR purpose.

File: templates/cpp-template-default/Classes/HelloWorldScene.h (hunk 1 above)
Comment (tarptaeya) [reply templates/cpp-template-default/Classes/HelloWorldScene.h:41]: But it gives warning in editor (QtCreator) and thus have to manually remove it.

Comment (jcs090218) [reply templates/cpp-template-default/Classes/HelloWorldScene.h:41]: Well, good to hear that. Then now this is the case I would say.

**Comment by stevetranby** [comment]
Interesting, I've never had any warnings on other IDE/compilers. 
It is a valid semicolon (for >= c++11).

It makes sense, pedantically, to remove it as it's unnecessary.

For some reason with c++ I do prefer the excess `;` though definitely mostly out of consistency/habit since `CREATE_FUNC` feels like a declaration and not a definition. I don't otherwise put a `;` after function bodies, unless they're single MACRO statements where it just feels weird without, even if correct. Heh.

**Comment by jcs090218** [comment]
@Tarptaeya I think you might need to provide a bit more info in order to reproduce the warning?

**Comment by tarptaeya** [comment]
@jcs090218 Just open the newly created project in the QtCreator, there it shows
```
Parser Issue
Warning: Extra ';' after member function definition
(-Wextra-semi)
```

**Comment by minggo** [comment]
extra `;` make it more readable, i think you should modify your IDE to remove the warning.

**Comment by Dimon4eg** [comment]
@minggo maybe just delete semicolon from macro?

**Comment by minggo** [comment]
The macro doesn't have `;`, it ends with `}`.

**Comment by Dimon4eg** [comment]
Oh sorry

**Pull request closed without merging**